server/speech.js

```diff
@@ -0,0 +1,119 @@
+// This endpoint handles server calls to get transcript of an audio file
+// We check the database for an existing transcript, and if we find none
+// we use the IBM Watson api to transcribe one from the wav file on s3 and 
+// store it
+function transcribeEndpoint(pool, request, response) {
+  var audioFile = request.headers['x-teachermoments-audio'];
+  console.log('audioFile from header:',audioFile);
+
+  //Hard coding this audioFile to a downloadUrl in my local database
+  //audioFile should be the downloadUrl for a wav file in s3
+  //Should not need this line when hooked to s3
+  audioFile = "blob:http://localhost:3000/a510a130-f17d-4331-90fe-1f2c8c0a2083";
+
+  const values = [audioFile];
+  const sql = `
+    SELECT transcript
+    FROM evidence
+    WHERE 1=1
+      AND json->>'downloadUrl' = $1
+    ORDER BY json->>'sessionId', id ASC;`;
+
+  pool.query(sql,values)
+    .then(results => {
+      if ((results.rowCount >= 1) && (results.rows[0].transcript !== "")){
+        console.log('found transcript: ', results.rows[0].transcript);
+        response.set('Content-Type', 'application/json');
+        response.json({
+          transcript: results.rows[0].transcript
+        });
+        return response.status(200).end();
+      }
+      else{
+        console.log('need to transcribe');
+        speechToText(pool,audioFile)
+          .then(results => {
+            console.log('final transcript: ',results);
+            response.set('Content-Type', 'application/json');
+            response.json({
+              transcript: results 
+            });
+            return response.status(500).end();
+          })
+          .catch(err => {
+            console.log('Error with transcription:', err);
+            return response.status(500).end();
+          });
+      }
+    })
+    .catch(err => {
+      console.log('Error with database query for transcript:', err);
+      return response.status(500).end();
+    });
+}
+
+// Sets up connection to IBM Watson api and transcribes audio to text.
+// returns a promise resolved with the transcript
+function speechToText(pool, audioFile) {
+  //changing audioFile here to a wav file on my local disk to transcribe
+  //Should not need this line when hooked up to s3
+  audioFile = '/Users/keving17/Documents/Github/TSL/teacher-moments/server/20180213-185953-485_8d44f67e-d783-4597-aec2-d417325cce06.wav';
+  
+  var SpeechToTextV1 = require('watson-developer-cloud/speech-to-text/v1');
+  var fs = require('fs');
+
+  var watson = new SpeechToTextV1 ({
```

Comment:
This is GitHub, so your credentials are now public :)  I'm sure this is fine for your hobby demo account, but in general you should store these kinds of things in environment variables.  Then this changes to `process.env.WATSON_USERNAME` and when you run the server locally you can do `WATSON_USERNAME=foo yarn start` or whatever.